# IS 362 - Week 5 Assignment

## What Is the Northernmost Airport in the United States?

To find the answer to this question the source data was read into pandas from the airpots.CSV file and a DataFrame was created. 

In [3]:
# import pandas
import pandas as pd

# load data from CSV file into a DataFrame
df = pd.read_csv("airports.csv")

# print DataFrame
df

faa                           name        lat         lon   alt  tz dst  \
0     04G              Lansdowne Airport  41.130472  -80.619583  1044  -5   A   
1     06A  Moton Field Municipal Airport  32.460572  -85.680028   264  -6   A   
2     06C            Schaumburg Regional  41.989341  -88.101243   801  -6   A   
3     06N                Randall Airport  41.431912  -74.391561   523  -5   A   
4     09J          Jekyll Island Airport  31.074472  -81.427778    11  -5   A   
...   ...                            ...        ...         ...   ...  ..  ..   
1453  ZUN                     Black Rock  35.083228 -108.791778  6454  -7   A   
1454  ZVE         New Haven Rail Station  41.298669  -72.925992     7  -5   A   
1455  ZWI      Wilmington Amtrak Station  39.736667  -75.551667     0  -5   A   
1456  ZWU       Washington Union Station  38.897460  -77.006430    76  -5   A   
1457  ZYP                   Penn Station  40.750500  -73.993500    35  -5   A   

                 tzone  
0     America/New_York  
1      America/Chicago  
2      America/Chicago  
3     America/New_York  
4     America/New_York  
...                ...  
1453    America/Denver  
1454  America/New_York  
1455  America/New_York  
1456  America/New_York  
1457  America/New_York  

[1458 rows x 8 columns]

To find the northernmost airport from the DataFrame, the lat column was sorted and the last 5 rows were selected.

In [4]:
# sorting the lat column (in ascending order)
northernmost = df.sort_values(by='lat')

# select the last 5 rows to get the top 5 northernmost airports
northernmost.tail()

faa                                        name        lat         lon  \
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
708  K03                               Wainwright As  70.613378 -159.860350   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   

     alt  tz dst              tzone  
152   96  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
230   44  -9   A  America/Anchorage  
417  149  -5   A                NaN

According to the dataset, Dillant Hopkins Airport is the northernmost airport in the United States. To confirm this I used Google Maps to verify the coordinates and found that the longitude and latitude values were swapped and the latitude for Dillant Hopkins Airport is actually 42.898333. I then went to the second highest value on the list to verify the coordinates and found the coordinates correct. So Wiley Post Will Rogers Mem is the northernmost airport in the United States. 

In [5]:
# select the second to last row of northernmost
northernmost.iloc[[-2]]

faa                        name        lat         lon  alt  tz dst  \
230  BRW  Wiley Post Will Rogers Mem  71.285446 -156.766003   44  -9   A   

                 tzone  
230  America/Anchorage

## 2.) What Is the Easternnmost Airport in the United States?
To find the easternmost airport from the DataFrame, the lon column was sorted and the last 5 rows were selected.

In [6]:
# sorting the lat column (in ascending order)
easternmost = df.sort_values(by='lon')

# select the last 5 rows to get the top 5 northernmost airports
easternmost.tail()

faa                           name        lat         lon   alt  tz dst  \
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   

                  tzone  
444    America/New_York  
417                 NaN  
396      Asia/Chongqing  
942      Asia/Chongqing  
1290  America/Anchorage

Excluding outliers, Eastport Municipal Airport is the easternmost airport in the United States.

In [7]:
# select the second to last row of easternnmost
easternmost.iloc[[-5]]

faa                        name        lat        lon  alt  tz dst  \
444  EPM  Eastport Municipal Airport  44.910111 -67.012694   45  -5   A   

                tzone  
444  America/New_York

## 3.) On February 12th, 2023, which New York area airport had the windiest weather?
To find the answer to this question the source data was read into pandas from the weather.CSV file and a DataFrame was created. 

In [10]:
# load data from CSV file into a DataFrame
df2 = pd.read_csv("weather.csv")

# print DataFrame
df2

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0        EWR  2013      1    1     1  39.02  26.06  59.37     270.0   
1        EWR  2013      1    1     2  39.02  26.96  61.63     250.0   
2        EWR  2013      1    1     3  39.02  28.04  64.43     240.0   
3        EWR  2013      1    1     4  39.92  28.04  62.21     250.0   
4        EWR  2013      1    1     5  39.02  28.04  64.43     260.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
26110    LGA  2013     12   30    14  35.96  19.94  51.78     340.0   
26111    LGA  2013     12   30    15  33.98  17.06  49.51     330.0   
26112    LGA  2013     12   30    16  32.00  15.08  49.19     340.0   
26113    LGA  2013     12   30    17  30.92  12.92  46.74     320.0   
26114    LGA  2013     12   30    18  28.94  10.94  46.41     330.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
0        10.35702        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1         8.05546        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        11.50780        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        12.65858        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        12.65858        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
26110    13.80936   21.86482     0.0    1017.1   10.0  2013-12-30T19:00:00Z  
26111    17.26170   21.86482     0.0    1018.8   10.0  2013-12-30T20:00:00Z  
26112    14.96014   23.01560     0.0    1019.5   10.0  2013-12-30T21:00:00Z  
26113    17.26170        NaN     0.0    1019.9   10.0  2013-12-30T22:00:00Z  
26114    18.41248        NaN     0.0    1020.9   10.0  2013-12-30T23:00:00Z  

[26115 rows x 15 columns]

Then I selected rows with the date 2/12/2013.

In [12]:
# select rows with date 2/12/2013
windiest = df2[(df2.year == 2013) & (df2.month == 2) & (df2.day == 12)]

# print DataFrame
windiest

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1007     EWR  2013      2   12     1  39.92  37.94  92.56     250.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
1010     EWR  2013      2   12     4  39.02  26.96  64.29     280.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
18434    LGA  2013      2   12    19  41.00  21.92  46.23     260.0   
18435    LGA  2013      2   12    20  39.92  23.00  50.46     290.0   
18436    LGA  2013      2   12    21  39.02  21.92  49.93     310.0   
18437    LGA  2013      2   12    22  39.02  23.00  52.26     310.0   
18438    LGA  2013      2   12    23  37.94  23.00  54.51     300.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1006      6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1007      9.20624        NaN     0.0    1007.2   10.0  2013-02-12T06:00:00Z  
1008     20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
1010     12.65858        NaN     0.0    1008.8   10.0  2013-02-12T09:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
18434    14.96014   21.86482     0.0    1014.7   10.0  2013-02-13T00:00:00Z  
18435    10.35702        NaN     0.0    1013.6   10.0  2013-02-13T01:00:00Z  
18436    10.35702        NaN     0.0    1012.8   10.0  2013-02-13T02:00:00Z  
18437    11.50780        NaN     0.0    1012.7   10.0  2013-02-13T03:00:00Z  
18438     9.20624        NaN     0.0    1013.5   10.0  2013-02-13T04:00:00Z  

[72 rows x 15 columns]

Then I ordered the windiest DataFrame by windspeed in ascending order and selected the last 5 rows. 

In [16]:
# order in ascending order
windiest_sorted = windiest.sort_values(by='wind_speed')

# print DataFrame
windiest_sorted.tail()


origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z

Excluding the outlier, the highest windspeed at a New York airport on February 12th was observed at LGA.

In [18]:
# select the second to last row of windiest_sorted
windiest_sorted.iloc[[-2]]

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
18417     23.0156   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z